### PR TITLE
Fix Citation modal responsiveness and clipboard copy

### PR DESCRIPTION
**Pull request by silverwind**
The modal was broken in two ways:

- On small screens, the input box was partially hanging outside the modal. Fixed with flexbox and increased modal width.
- The clipboard copy was not working because the modal had both `data-clipboard-text` and `data-clipboard-target`, while we only support one of those. Made a small tweak in clipboard as well so that it will still fall back to target if text is empty.

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: templates/repo/cite/cite_buttons.tmpl [hunk 1]
```diff
@@ -6,6 +6,6 @@ BibTeX
 </button>
 <!-- the value will be updated by initCitationFileCopyContent, the code below is used to avoid UI flicking  -->
 <input id="citation-copy-content" value="" size="1" readonly>
-<button class="ui icon button" id="citation-clipboard-btn" data-tooltip-content="{{ctx.Locale.Tr "copy"}}" data-clipboard-text="" data-clipboard-target="#citation-copy-content">
+<button class="ui icon button" id="citation-clipboard-btn" data-tooltip-content="{{ctx.Locale.Tr "copy"}}"  data-clipboard-target="#citation-copy-content">
```
Comment (silverwind) [inline templates/repo/cite/cite_buttons.tmpl]:
```suggestion
<button class="ui icon button" id="citation-clipboard-btn" data-tooltip-content="{{ctx.Locale.Tr "copy"}}" data-clipboard-target="#citation-copy-content">
```

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I do not think this could be backported, some code is 1.22-only.

**Comment by silverwind** [comment]
It should theoretically apply to 1.21, not many changes around these parts. But yeah I'm not thrilled about another manual backport. Let's still see whether GiteaBot can do it.

**Comment by wxiaoguang** [comment]
It can't, because I am sure the "copy" code was changed by #29379 , it is only in 1.22

**Comment by silverwind** [comment]
Ah I see, https://github.com/go-gitea/gitea/pull/29379 had regressed this copy button. The style fix is barely worth backporting imho.

**Pull request merged**